test/bm_serial_ut.cpp

```diff
@@ -351,3 +351,44 @@ TEST_F(NCPTest, ConfigTest) {
   EXPECT_TRUE(fake_cfg_key_del_response_fn_called);
 
 }
+
+static bool fake_network_info_fn_called;
+static bool fake_network_info_fn(bm_common_network_info_t* network_info) {
+  (void) network_info;
+  fake_network_info_fn_called = true;
+  return true;
+}
+
+TEST_F(NCPTest, NetworkInfoTest) {
+  _callbacks.tx_fn = fake_tx_fn;
+  _callbacks.network_info_fn = fake_network_info_fn;
+  bm_serial_set_callbacks(&_callbacks);
+  fake_network_info_fn_called = false;
+
+  bm_common_config_crc_t config_crc = {
+    .partition = BM_COMMON_CFG_PARTITION_SYSTEM,
+    .crc32 = 1234,
+  };
+
+  bm_common_fw_version_t fw_info = {
+    .major = 1,
+    .minor = 2,
+    .revision = 3,
+    .gitSHA = 1234,
+  };
+
+  uint64_t node_list[] = {12345678};
+  uint16_t num_nodes = 1;
+
+  uint16_t total_len = sizeof(bm_common_network_info_t) + num_nodes*sizeof(uint64_t);
+  bm_common_network_info_t *_network_info = (bm_common_network_info_t*)malloc(total_len);
+  _network_info->newtwork_crc32 = 1234;
```

Comment:
May be nice to make a convenience function like `bm_serial_pub` or `bm_serial_set_rtc`
to handle all the message packing code

Reply:
added in latest push, lmk what you think

Reply:
I'm actually going to change it... we need one to pack the message together then one to send it.